mod-source-record-storage-server/src/main/java/org/folio/services/handlers/actions/AbstractUpdateModifyEventHandler.java

```diff
@@ -80,17 +81,27 @@ public CompletableFuture<DataImportEventPayload> handle(DataImportEventPayload p
       String userId = (String) payload.getAdditionalProperties().get(USER_ID_HEADER);
       preparePayload(payload);
 
+      var eventContext = payload.getContext();
+      String entityAsString = eventContext.get(getMatchedMarcKey());
+      String recordAsString = getRecordAsString(payload, marcMappingOption);
+
       mappingParametersCache.get(payload.getJobExecutionId(), RestUtil.retrieveOkapiConnectionParams(payload, vertx))
         .compose(parametersOptional -> parametersOptional
           .map(mappingParams -> modifyRecord(payload, mappingProfile, mappingParams))
           .orElseGet(() -> Future.failedFuture(format(MAPPING_PARAMETERS_NOT_FOUND_MSG, payload.getJobExecutionId()))))
         .onSuccess(v -> prepareModificationResult(payload, marcMappingOption))
         .map(v -> Json.decodeValue(payloadContext.get(modifiedEntityType().value()), Record.class))
         .onSuccess(changedRecord -> {
-          if(isHridFillingNeeded() || isUpdateOption(marcMappingOption)) {
+          if (isHridFillingNeeded()) {
             addControlledFieldToMarcRecord(changedRecord, HR_ID_FROM_FIELD, hrId, true);
             remove003FieldIfNeeded(changedRecord, hrId);
           }
+          if (isUpdateOption(marcMappingOption)) {
+            Record record = Json.decodeValue(recordAsString, Record.class);
+            JsonObject externalEntity = new JsonObject(entityAsString);
+
+            fillHrIdFieldInMarcRecord(Pair.of(record, externalEntity), true);
```

Comment:
looks like fillHrIdFieldInMarcRecord() method does nothing because it expects pair of record and instance but here `externalEntity` is a matched marc record. Since a record does not have hrid, hrid [here ](https://github.com/folio-org/mod-source-record-storage/blob/767a2825c8a6bfcdd835e7dd2564d35854f7d1d8/mod-source-record-storage-server/src/main/java/org/folio/services/util/AdditionalFieldsUtil.java#L298) always will be null and therefore any `if` blocks will not be executed in the  fillHrIdFieldInMarcRecord() method.